pkg/didcomm/transport/http/inbound.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package http
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// InboundDIDCommHandler struct to pass details for request handlers
+type InboundDIDCommHandler struct {
+	Path        string
+	HandlerFunc http.HandlerFunc
+}
+
+// NewInboundHandler will create a new handler to enforce Did-Comm HTTP transport specs
+// then routes processing for the passed 'inURL' and 'handlerFunc' arguments.
+// Both 'inURL' and 'handlerFunc' are mandatory for inbound communications.
+func NewInboundHandler(inURL string, handlerFunc http.HandlerFunc) (http.Handler, error) {
```

Comment:
Your previous version of this PR had an option style argument. Are you saying that these two arguments are now always needed (and as this type)?

Reply:
yes, there needs to be 1 service endpoint and 1 http handler for inbound communication to happen as discussed with @llorllale

Reply:
@Baha-sk `inURL` is not required. Let the user worry about that in their layer.

`handlerFunc` should not be an `http.HandlerFunc` - it should be a custom interface that accepts a decrypted blob (ie. `[]byte`). This custom interface will be aware of the different DIDComm messages and know how to dispatch them

Reply:
removed